langstream-agents/langstream-agent-camel/src/main/java/ai/langstream/agents/camel/CamelSource.java

```diff
@@ -0,0 +1,249 @@
+        }
+    }
+
+    private class LangStreamProcessor extends AsyncProcessorSupport {
+
+        @Override
+        public boolean process(Exchange exchange, AsyncCallback callback) {
+
+            log.info("Processing exchange: {}", exchange);
+            Message in = exchange.getIn();
+            try {
+                Collection<Header> headers = new ArrayList<>();
+                if (in.hasHeaders()) {
+                    for (Map.Entry<String, Object> header : in.getHeaders().entrySet()) {
+                        String k = header.getKey();
+                        Object v = header.getValue();
+                        Object converted = safeObject(v);
+                        headers.add(new SimpleRecord.SimpleHeader(k, converted));
+                    }
+                    ;
+                }
+                Object key = keyHeader.isEmpty() ? null : safeObject(in.getHeader(keyHeader));
+                CamelRecord record =
+                        new CamelRecord(
+                                callback,
+                                exchange,
+                                key,
+                                safeObject(in.getBody()),
+                                componentUri,
+                                in.getMessageTimestamp(),
+                                headers);
+                log.info("Created record {}", record);
+                records.add(record);
+            } catch (Exception error) {
+                log.error("Error serializing record", error);
+                exchange.setException(error);
+                callback.done(true);
+            }
+
+            return true;
+        }
+
+        @Nullable
+        private static Object safeObject(Object v) throws JsonProcessingException {
+            Object converted;
+            if (v == null) {
+                converted = null;
+            } else if (v instanceof CharSequence || v instanceof Number || v instanceof Boolean) {
+                converted = v;
+            } else {
+                converted = MAPPER.writeValueAsString(v);
+            }
+            return converted;
+        }
+    }
+
+    @Override
+    public void init(Map<String, Object> configuration) throws Exception {
+        componentUri = ConfigurationUtils.getString("component-uri", "", configuration);
+        int maxRecords = ConfigurationUtils.getInt("max-buffered-records", 100, configuration);
+        this.keyHeader = ConfigurationUtils.getString("key-header", "", configuration);
+        Map<String, Object> componentOptions =
+                ConfigurationUtils.getMap("component-options", Map.of(), configuration);
+        if (componentOptions != null) {
+            for (Map.Entry<String, Object> entry : componentOptions.entrySet()) {
+                String paramName = entry.getKey();
+                Object value = entry.getValue();
+                if (value != null) {
+                    if (componentUri.contains("?")) {
```

Comment:
Could this be done with `java.net.URI` ?

Reply:
I don't know if Camel components URI are real URIs, IIRC there are some tricks, like RAW(xxx)